bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ErrorValue.java

```diff
@@ -302,9 +302,20 @@ public String getPrintableStackTrace() {
         for (int i = 1; i < stackTrace.length; i++) {
             printStackElement(sb, stackTrace[i], "\n\t   ");
         }
+        addErrorCauseWithLocation(sb, this.cause);
         return sb.toString();
     }
 
+    public void addErrorCauseWithLocation(StringBuilder sb, BError cause) {
+        if (cause != null) {
+            sb.append("\ncause: ")
+                    .append(cause.getMessage())
```

Comment:
Shall we try to reduce this code from getPrintableStackTrace. May be having common method.